cpp/src/arrow/util/bit-util.h

```diff
@@ -68,9 +68,13 @@ static inline void SetBit(uint8_t* bits, int64_t i) {
 }
 
 static inline void SetBitTo(uint8_t* bits, int64_t i, bool bit_is_set) {
-  // See https://graphics.stanford.edu/~seander/bithacks.html
+  // TODO: speed up. See https://graphics.stanford.edu/~seander/bithacks.html
   // "Conditionally set or clear bits without branching"
-  bits[i / 8] ^= static_cast<uint8_t>(-bit_is_set ^ bits[i / 8]) & kBitmask[i % 8];
+  if (bit_is_set) {
```

Comment:
We'll probably also want to start a util-benchmark.cc when working on this so we aren't guessing about what's more performant (the Stanford link claims 5-15% faster without branching, but it would be good to know for sure)